### PR TITLE
Created new persistence algorithm

### PR DESCRIPTION
**Pull request by Bartechnika**
### **Describe your change:**
Added a persistence algorithm that determines the persistence of a number.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/persistence.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    steps = 0
+    num_string = str(num)
+
+    while len(num_string) != 1:
+        numbers = [int(i) for i in num_string]
+
+        total = 1
+        for i in range(0, len(numbers)):
+            total *= numbers[i]
```
[the remark this answers is not in the document]
Comment (poyea) [reply maths/persistence.py:32]: Let's stick with the reference? But it seems like this will also work.

[the remark this answers is not in the document]
Comment (Bartechnika) [reply maths/persistence.py:32]: Sounds good, I will update the program to use` total += numbers[i]`.

[the remark this answers is not in the document]
Comment (Bartechnika) [reply maths/persistence.py:32]: Okay, It states in the Wikipedia article that are 2 types of persistence. Originally, I made this to use multiplicative persistence, but additional persistence is also a thing. I can make this program use both methods, or I can make a separate program for each? Sorry for the confusion, thanks again.

[the remark this answers is not in the document]
Comment (poyea) [reply maths/persistence.py:32]: You can make a separate function in this file for addtion.

[the remark this answers is not in the document]
Comment (Bartechnika) [reply maths/persistence.py:32]: > You can make a separate function in this file for addtion.
Thankyou so much thats a great idea.

[the remark this answers is not in the document]
Comment (Bartechnika) [reply maths/persistence.py:32]: All done, should be working now. Thanks for the help!

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**